### DISCUSSION TITLE
Celeron G3900

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#253] [General]
@svmlegacy
Thank you for your Celeron addition in the Wiki

There are things to learn from, like: 
* HWP showing zero values although it is enabled
* Missing extended DRAM timings
* No MWAIT/X although it is  Skylake

I suggest you try latest version or `develop` branch. Fixes are available.

**Comment by svmlegacy** [comment]
I'll get a proper install of a distro on this machine to accomodate. This was using the Arch Linux bootable ISO as it doesn't currently have a disk

**Comment by svmlegacy** [comment]
Celeron G3900
Arch Linux Bootable ISO [here](https://gist.github.com/svmlegacy/aed26afa533939b202af069b79ef5bfa)
1.86.0 master branch [here](https://gist.github.com/svmlegacy/a864e8fb0253aa0461e1dc16172e2560)
1.86.2 develop branch [here](https://gist.github.com/svmlegacy/95c8fe827dab5cce7a8d1bf518251194)

Wiki G3900 link updated to 1.86.0 master branch

**Reply by cyring** [reply]
Between old version within ISO and latest master or develop, I don't read same DRAM main timings:
15-15-15 vs 16-16-16
I wonder if it is a code  regression or improvement.
Which timings do you believe is accurate ?

**Reply by svmlegacy** [reply]
I believe both are accurate, as the memory configuration changed between the ISO and the latest versions.
At 2133 MHz the first module I used is CL15, and the added module is CL16 at the same speed.

Seems like the motherboard is picking the looser of the two, and running both at CL16.

**Comment by cyring** [comment]
Issue: DIMM geometry can not be the same with 8 or 4 GB stick
Fortunately some bits provide us  the DIMM size but the standard equation is not verified:
```
8 × Bank × Rank × Rows × Columns = Size
```
